ModelLibrary/Process/MasterForecast.cs

```diff
@@ -32,20 +32,37 @@ public class MasterForecast : SvrProcess
         string sql = "";
         int C_Period_ID = 0;
         X_C_MasterForecast mf = null;
+        DataSet dsOrder = null;
+        DataSet dsForecast = null;
+        DataSet dsOpp = null;
+        MMasterForecastLine mfLine = null;
+        int Count = 0;
+        int tableId = 0;
+        MTable tbl = null;
+        PO po = null;
         #endregion
-
         protected override void Prepare()
         {
 
         }
 
+        /// <summary>
+        /// Consolidate Data from sales order , Team Forecast, Opportunity
+        /// </summary>
+        /// <returns>Info</returns>
         protected override string DoIt()
         {
-            mf = new X_C_MasterForecast(GetCtx(), GetRecord_ID(), null);
+            mf = new X_C_MasterForecast(GetCtx(), GetRecord_ID(), Get_Trx());
             C_Period_ID = mf.GetC_Period_ID();
 
-           // sql = "delete from c_masterforecastline where c_masterforecast_id = " + mf.GetC_MasterForecast_ID();
-           // int count = DB.ExecuteQuery(sql, null, null);
+            //Get Table_Id to create PO Object
+            sql = @"SELECT AD_TABLE_ID  FROM AD_TABLE WHERE tablename LIKE 'VA073_MasterForecastLineDetail' AND IsActive = 'Y'";
+            tableId = Util.GetValueOfInt(DB.ExecuteScalar(sql, null, null));
+            tbl = new MTable(GetCtx(), tableId, null);
```

Comment:
Or you can use this for getting table Object :: public static MTable Get(Ctx ctx, string tableName)